04-radio4000.sql

```diff
@@ -135,10 +136,54 @@ $$;
 -- the trigger will set the "updated_at" column to the current timestamp for every update
 create extension if not exists moddatetime schema extensions;
 create trigger user_update before update on users
-  for each row execute procedure moddatetime (updated_at);
+	for each row execute procedure moddatetime (updated_at);
 create trigger channel_update before update on channels
-  for each row execute procedure moddatetime (updated_at);
+	for each row execute procedure moddatetime (updated_at);
 create trigger user_channel_update before update on user_channel
-  for each row execute procedure moddatetime (updated_at);
+	for each row execute procedure moddatetime (updated_at);
 create trigger channel_track_update before update on channel_track
-  for each row execute procedure moddatetime (updated_at);
+	for each row execute procedure moddatetime (updated_at);
+
+-- Usage: parse_tokens(myString, '#')
+CREATE or replace FUNCTION parse_tokens(content text, prefix text)  
+	RETURNS text[]
+	LANGUAGE plpgsql
+AS $$
+	DECLARE
+		regex text;
+		matches text;
+		subquery text;
+		captures text;
+		tokens text[];
+	BEGIN
+		regex := prefix || '(\S+)';
+		matches := 'regexp_matches($1, $2, $3) as captures';
+		subquery := '(SELECT ' || matches || ' ORDER BY captures) as matches';
+		captures := 'array_agg(matches.captures[1])';
+
+		EXECUTE 'SELECT ' || captures || ' FROM ' || subquery
+		INTO tokens
+		USING LOWER(content), regex, 'g';
+
+		IF tokens IS NULL THEN
+			tokens = '{}';
+		END IF;
+
+		RETURN tokens;
+	END;
+$$;
+
+create or replace function parse_track_tags()
+	returns trigger
+	language plpgsql
+as $$
+	begin
+		new.tags = parse_tokens(new.description, '#');
+		-- new.mentions = parse_tokens(new.description, '@');
```

Comment:
Do we want to add "mentions" right away? Maybe not.

Reply:
good question! haven't though of that.
we can have them to try?

Reply:
done!